src/Site.js

```diff
@@ -385,6 +412,193 @@ Site.prototype.createPage = function (config) {
+      .then(() => this.addDefaultLayoutToSiteConfig(rootPath))
+      .then(() => this.addTopNavToAddressablePages(rootPath))
+      .then(resolve)
+      .catch(reject);
+  });
+};
+
+/**
+ * Adds an index page to site if not present.
+ * @param rootPath
+ */
+Site.prototype.addIndexPage = function (rootPath) {
+  const indexPagePath = path.join(rootPath, INDEX_MARKDOWN_FILE);
+  const fileNames = ['README.md', 'Home.md'];
+  for (let i = 0; i < fileNames.length; i += 1) {
+    const filePath = path.join(rootPath, fileNames[i]);
+    if (fs.existsSync(filePath)) {
+      fs.copyAsync(filePath, indexPagePath)
+        .catch(() => Promise.reject(new Error(`Failed to copy over ${fileNames[i]}`)));
+      break;
+    }
+  }
+  return Promise.resolve();
+};
+
+/**
+ * Adds an about page to site if not present.
+ * @param rootPath
+ */
+Site.prototype.addAboutPage = function (rootPath) {
+  const aboutPath = path.join(rootPath, ABOUT_MARKDOWN_FILE);
+  return new Promise((resolve, reject) => {
+    fs.accessAsync(aboutPath)
+      .catch(() => {
+        if (fs.existsSync(aboutPath)) {
+          return Promise.resolve();
+        }
+        return fs.outputFileAsync(aboutPath, ABOUT_MARKDOWN_DEFAULT);
+      })
+      .then(resolve)
+      .catch(reject);
+  });
+};
+
+/**
+ * Adds a footer to default layout of site.
+ * @param rootPath
+ */
+Site.prototype.addFooterToDefaultLayout = function (rootPath) {
+  const footerPath = path.join(rootPath, FOOTER_PATH);
+  const siteLayoutPath = path.join(rootPath, LAYOUT_FOLDER_PATH);
+  const siteLayoutFooterDefaultPath = path.join(siteLayoutPath, LAYOUT_DEFAULT_NAME, 'footer.md');
+  const wikiFooterPath = path.join(rootPath, WIKI_FOOTER_PATH);
+
+  return new Promise((resolve, reject) => {
+    fs.accessAsync(wikiFooterPath)
+      .then(() => {
+        const footerContent = fs.readFileSync(wikiFooterPath, 'utf8');
+        const wrappedFooterContent = `<footer>\n\t${footerContent}\n</footer>`;
+        return fs.outputFileSync(siteLayoutFooterDefaultPath, wrappedFooterContent);
+      })
+      .catch(() => {
+        if (fs.existsSync(footerPath)) {
+          return fs.copyAsync(footerPath, siteLayoutFooterDefaultPath);
+        }
+        return Promise.resolve();
+      })
+      .then(resolve)
+      .then(reject);
+  });
+};
+
+/**
+ * Adds a site navigation bar to the default layout of the site.
+ * @param rootPath
+ */
+Site.prototype.addSiteNavToDefaultLayout = function (rootPath) {
+  const siteLayoutPath = path.join(rootPath, LAYOUT_FOLDER_PATH);
+  const siteLayoutSiteNavDefaultPath = path.join(siteLayoutPath, LAYOUT_DEFAULT_NAME, 'navigation.md');
+  const wikiSiteNavPath = path.join(rootPath, WIKI_SITE_NAV_PATH);
+
+  return new Promise((resolve, reject) => {
+    fs.accessAsync(wikiSiteNavPath)
+      .then(() => {
+        const siteNavContent = fs.readFileSync(wikiSiteNavPath, 'utf8');
+        const wrappedSiteNavContent = `<navigation>\n${siteNavContent}\n</navigation>`;
```

Comment:
For wiki's with side bars the urls are not resolved correctly:

https://github.com/SFML/SFML/wiki

![image](https://user-images.githubusercontent.com/22221132/53693368-74028d80-3dda-11e9-9dd1-f2cdc41960c4.png)

Reply:
Also the styling seems to be off to the side for both types of generation of site navs

Reply:
> For wiki's with sidebars the urls are not resolved correctly:
You're right, I think I might need to preprocess `_Sidebar.md` instead of just copying it over.

> Also the styling seems to be off to the side for both types of generation of site navs
If I'm not wrong, it's because of a bug that's mentioned in #739.